language/move-lang/src/parser/syntax.rs

```diff
@@ -98,6 +98,31 @@ fn consume_optional_token_with_loc<'input>(
     }
 }
 
+fn parse_comma_list<'input, F, R>(
```

Comment:
Just a reminder that later when you run into parsing << and >>, you may want to make this even more generic. 

This is what I had to do for the IR parser:
```
fn parse_comma_list<'input, F, G, R>(
    tokens: &mut Lexer<'input>,
    is_end_of_list: G,
    parse_list_item: F,
    allow_trailing_comma: bool,
) -> Result<Vec<R>, ParseError<usize, anyhow::Error>>
where
    F: Fn(&mut Lexer<'input>) -> Result<R, ParseError<usize, anyhow::Error>>,
    G: Fn(&Lexer<'input>) -> bool,
{
    let mut v = vec![];
    if !is_end_of_list(tokens) {
        loop {
            v.push(parse_list_item(tokens)?);
            if is_end_of_list(tokens) {
                break;
            }
            consume_token(tokens, Tok::Comma)?;
            if is_end_of_list(tokens) && allow_trailing_comma {
                break;
            }
        }
    }
    Ok(v)
}
```

Reply:
I'm planning to add an error message saying "expected '{}'" for the list end token, so using an "is_end_of_list" function won't work